tests/common/tests.rs

```diff
@@ -916,3 +916,526 @@ async fn variable_auth_live_query() -> Result<(), Box<dyn std::error::Error>> {
+				SIGNIN ( SELECT * FROM user WHERE email = $email AND crypto::argon2::compare(pass, $pass) )
+			;"#,
+		)
+		.await?;
+	// Create resource that requires a scope session to query
+	socket
+		.send_message_query(
+			r#"
+			DEFINE TABLE test SCHEMALESS
+				PERMISSIONS FOR select, create, update, delete WHERE $scope = "scope"
+			;"#,
+		)
+		.await?;
+	socket
+		.send_message_query(
+			r#"
+			CREATE test:1 SET working = "yes"
+			;"#,
+		)
+		.await?;
+	// Send SIGNUP command
+	let res = socket
+		.send_request(
+			"signup",
+			json!(
+				[{
+					"ns": NS,
+					"db": DB,
+					"sc": "scope",
+					"email": "email@email.com",
+					"pass": "pass",
+				}]
+			),
+		)
+		.await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	// Verify response contains no error
+	assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+	// Verify it returns a token
+	assert!(res["result"].is_string(), "result: {:?}", res);
+	let res = res["result"].as_str().unwrap();
+	assert!(res.starts_with("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9"), "result: {}", res);
+	// Authenticate using the token, which will expire soon
+	socket.send_request("authenticate", json!([res,])).await?;
+	// Check if the session is now authenticated
+	let res = socket.send_message_query("SELECT VALUE working FROM test:1").await?;
+	assert_eq!(res[0]["result"], json!(["yes"]), "result: {:?}", res);
+	// Wait two seconds for token to expire
+	tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+	// Check that the session has expired and queries fail
+	let res = socket.send_request("query", json!(["SELECT VALUE working FROM test:1",])).await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	assert_eq!(res["error"], json!({"code": -32000, "message": "There was a problem with the database: The session has expired"}));
+	// Sign in again using the same session
+	let res = socket
+		.send_request(
+			"signin",
+			json!(
+				[{
+					"ns": NS,
+					"db": DB,
+					"sc": "scope",
+					"email": "email@email.com",
+					"pass": "pass",
+				}]
+			),
+		)
+		.await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	// Verify response contains no error
+	assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+	// Check that the session is now valid again and queries succeed
+	let res = socket.send_message_query("SELECT VALUE working FROM test:1").await?;
+	assert_eq!(res[0]["result"], json!(["yes"]), "result: {:?}", res);
+	// Test passed
+	server.finish();
+	Ok(())
+}
+
+#[test(tokio::test)]
+async fn session_expiration_operations() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
https://github.com/surrealdb/surrealdb/pull/3561/files#r1500809184

Reply:
Resolved in [213e158](https://github.com/surrealdb/surrealdb/pull/3561/commits/213e1585edbb87540b6871b830f7f999f80430b2).